pkg/topology/full/full.go

```diff
@@ -130,6 +133,34 @@ func (d *Driver) ChunkPeer(addr swarm.Address) (peerAddr swarm.Address, err erro
 	return swarm.Address{}, topology.ErrNotFound
 }
 
+func (d *Driver) SyncPeer(addr swarm.Address) (peerAddr swarm.Address, err error) {
+	connectedPeers := d.p2pService.Peers()
+	if len(connectedPeers) == 0 {
+		return swarm.Address{}, topology.ErrNotFound
+	}
+
+	cpeer := d.base // start checking closest from _self_
+	for _, peer := range connectedPeers {
+		switch d := swarm.ProxCmp(addr.Bytes(), cpeer.Bytes(), peer.Address.Bytes()); d {
```

Comment:
I find it more readable to split such in two lines. Feel free to ignore this message.